addons/mail/models/mail_thread.py

```diff
@@ -2262,13 +2262,21 @@ def _notify_record_by_email(self, message, recipients_data, msg_vals=False,
         mail_subject = message.subject or (message.record_name and 'Re: %s' % message.record_name) # in cache, no queries
         # Replace new lines by spaces to conform to email headers requirements
         mail_subject = ' '.join((mail_subject or '').splitlines())
+        # compute references: set references to the parent and add current message just to
+        # have a fallback in case replies mess with Messsage-Id in the In-Reply-To (see
+        # task-2643114)
+        message_sudo = message.sudo()
+        if message_sudo.parent_id:
+            references = f'{message_sudo.parent_id.message_id} {message_sudo.message_id}'
```

Comment:
I see here:
https://github.com/odoo/odoo/blob/13.0/addons/mail/models/mail_thread.py#L1483

That we only look for one parent_id (with the `limit=1`).
But with this change there can be multiple parent_id within the references.

Maybe it's ok because we just want to find one parent, and it doesn't matter which one it is, just checking as I don't know that part of the code very well.

(Would be nice to adapt tests that parse messages to include samples with multiple parent_id within the references, if such tests exist, or maybe it's already tested?).

Reply:
Added some tests, this highlighted another issue ... so another fix in this branch. Could you re-check :D ?

Reply:
maybe we can add a order="create_date DESC" in https://github.com/odoo/odoo/blob/13.0/addons/mail/models/mail_thread.py#L1483 to be sure to take the newest as parent ?

Reply:
Done